app/screens/join_team/join_team.tsx

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {Alert, View} from 'react-native';
+
+import {addCurrentUserToTeam, fetchAllTeams, handleTeamChange} from '@actions/remote/team';
+import {PER_PAGE_DEFAULT} from '@client/rest/constants';
+import FormattedText from '@components/formatted_text';
+import Empty from '@components/illustrations/no_team';
+import Loading from '@components/loading';
+import TeamList from '@components/team_list';
+import {ServerErrors} from '@constants';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import useNavButtonPressed from '@hooks/navigation_button_pressed';
+import {dismissModal} from '@screens/navigation';
+import {isServerError} from '@utils/errors';
+import {logDebug} from '@utils/log';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+type Props = {
+    joinedIds: Set<string>;
+    componentId: string;
+    closeButtonId: string;
+}
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    container: {
+        paddingHorizontal: 10,
+        paddingVertical: 5,
+        flex: 1,
+    },
+    empty: {
+        flex: 1,
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+    loading: {
+        flex: 1,
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+    title: {
+        color: theme.centerChannelColor,
+        lineHeight: 28,
+        marginTop: 16,
+        ...typography('Heading', 400, 'Regular'),
+    },
+    description: {
+        color: theme.centerChannelColor,
+        marginTop: 8,
+        maxWidth: 334,
+        ...typography('Body', 200, 'Regular'),
+    },
+}));
+
+const LOAD_MORE_THRESHOLD = 10;
+
+export default function JoinTeam({
+    joinedIds,
+    componentId,
+    closeButtonId,
+}: Props) {
+    const serverUrl = useServerUrl();
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+    const intl = useIntl();
+    const page = useRef(0);
+    const hasMore = useRef(true);
+    const mounted = useRef(true);
+    const [loading, setLoading] = useState(true);
+    const [joining, setJoining] = useState(false);
+    const [otherTeams, setOtherTeams] = useState<Team[]>([]);
+
+    const loadTeams = useCallback(async (alreadyLoaded = 0) => {
```

Comment:
Nit: this function could also be extracted because it is used in both `join_team.tsx` and `select_team.tsx` with the exception of `setOtherTeams()` using teams vs noJoinedTeams